transport-sctp/src/main/java/io/netty/channel/sctp/SctpMessage.java

```diff
@@ -142,6 +142,7 @@ public boolean equals(Object o) {
     public int hashCode() {
         int result = streamIdentifier;
         result = 31 * result + protocolIdentifier;
+        result = 31 * result + Boolean.hashCode(unordered);
```

Comment:
That's java8 only o we can not use it. I think something like this would be good enough:

```java
result = 31 * result + (unordered ? 1231 : 1237);
```

@jamel could you just adjust your patch to this. Maybe also adding a comment that the numbers are these that are referenced in the javadocs of Boolean#hashCode().

See https://docs.oracle.com/javase/6/docs/api/java/lang/Boolean.html#hashCode()